src/datasets/utils/download_manager.py

```diff
@@ -64,6 +71,20 @@ class GenerateMode(enum.Enum):
     FORCE_REDOWNLOAD = "force_redownload"
 
 
+def _get_extraction_protocol_local(path: str) -> Optional[str]:
```

Comment:
A nit for consistency with `streaming_download_manager.py`:
```suggestion
def _get_extraction_protocol(path: str) -> Optional[str]:
```

Reply:
Hello @mariosasko @lhoestq  Thanks for the answers, I wasn't active lately I will get back to complete this pull request later this week.

Reply:
thanks @Mehdi2402 !